### PR TITLE
[Chore] Integrate external near-account-id.

### PR DESCRIPTION
**Pull request by ruseinov**
TODO: Switch near-account-id dependency to `near` repo as opposed to my personal one, once the `len` PR is merged.

### CODE REVIEW

File: chain/network/src/store/schema/mod.rs [hunk 1]
```diff
@@ -20,7 +20,8 @@ pub struct AccountIdFormat;
 impl Format for AccountIdFormat {
     type T = AccountId;
     fn encode<W: io::Write>(a: &AccountId, w: &mut W) -> io::Result<()> {
-        w.write_all(a.as_ref().as_bytes())
+        let account_id_ref: &AccountIdRef = a.as_ref();
+        w.write_all(account_id_ref.as_bytes())
```
Comment (frol) [inline chain/network/src/store/schema/mod.rs:24]: Try this one:

```suggestion
        w.write_all(a.as_bytes())
```

If it works ([it should](https://github.com/near/near-account-id/blob/main/src/account_id_ref.rs#L76)), apply everywhere else

File: runtime/runtime/src/verifier.rs [hunk 2]
```diff
@@ -238,7 +239,8 @@ pub fn verify_and_charge_transaction(
                 )
                 .into());
             }
-            if transaction.receiver_id.as_ref() != function_call_permission.receiver_id {
```
Comment (frol) [inline runtime/runtime/src/verifier.rs:241]: You probably don't need `as_ref()` here now at all as there is comparison implementations for AccountIdRef <> str

**Review by frol — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -174,7 +172,7 @@ lru = "0.7.2"
 memmap2 = "0.5"
 memoffset = "0.8"
 more-asserts = "0.2"
-near-account-id = { path = "core/account-id", features = ["internal_unstable"] }
+near-account-id = { git = "https://github.com/ruseinov/near-account-id", features = ["internal_unstable", "serde", "borsh"], branch = "ru/chore/len" }
```
Comment (nagisa) [inline Cargo.toml:175]: Why not make the relevant change in this PR itself?

**Comment by nagisa** [comment]
I would like to see the PR description contain the background information on this change – the motivation behind why this change is being made, references to discussion that has taken (if any – so far we've been moving in the direction of making it more monorepo rather than the other way around,) etc.

**Comment by frol** [comment]
@nagisa near-account-id was created inside nearcore as an attempt to easily reuse this crate in other parts of ecosystem (near-sdk-rs, near-cli-rs, near-jsonrpc-client-rs, …), but with the time passing, I believe that we can release this crate as 1.0 https://github.com/near/near-account-id-rs/issues/5 and extract it from the nearcore 0.x releases.

There are more such 1.0 crates will appear:
* https://github.com/near/near-gas-rs
* https://github.com/near/near-token-rs

This all is part of a bigger initiative to finally make the tooling landscape easy for consumption in on-chain and off-chain apps

File: chain/network/src/store/schema/mod.rs [hunk 4]
```diff
@@ -5,7 +5,7 @@ use crate::types as primitives;
 use borsh::{BorshDeserialize, BorshSerialize};
 use near_async::time;
 use near_crypto::Signature;
-use near_primitives::account::id::AccountId;
+use near_primitives::account::id::{AccountId, AccountIdRef};
```
Comment (frol) [inline chain/network/src/store/schema/mod.rs:8]: `AccountIdRef` imports are not needed anymore.

**Review by frol — CHANGES_REQUESTED** [review]

**Comment by ruseinov** [comment]
@frol Some of the workflows need to be approved. The error I'm getting now is kinda strange..

> @nagisa near-account-id was created inside nearcore as an attempt to easily reuse this crate in other parts of ecosystem (near-sdk-rs, near-cli-rs, near-jsonrpc-client-rs, …), but with the time passing, I believe that we can release this crate as 1.0 [near/near-account-id-rs#5](https://github.com/near/near-account-id-rs/issues/5) and extract it from the nearcore 0.x releases.
> 
> There are more such 1.0 crates will appear:
> 
> * https://github.com/near/near-gas-rs
> * https://github.com/near/near-token-rs
> 
> This all is part of a bigger initiative to finally make the tooling landscape easy for consumption in on-chain and off-chain apps

**Pull request closed without merging**